src/main/java/net/minecraftforge/common/ToolActions.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common;
+
+import java.util.List;
+
+public class ToolActions
```

Comment:
Should this class be similar to how ToolType was in earlier versions for if modders want to define custom actions that then there is a way to get all the actions and also could use instance equality for checking the actions.

Reply:
I'm open for it. I see both options as valid.

Reply:
Yes, I've already told him he should model this after tool types.
Basically making it a manual enum. Static final ToolAction fields, and a get factory.
It's basically the same as string.intern() but having defined fields is more likely to cause compile errors if you typo the action name. Or if you're making a custom one, it's only one place you need to fix not 100.

Reply:
It is done. :P